Pichu changes

H3ra-57 opened the pull request:
A big batch of Pichu changes. Might end up pretty overpowered, so while you should focus on stability first, comments on balance are appreciated.
Changelog:
[Pichu.Changelog.txt](https://github.com/HDR-Development/HewDraw-Remix/files/8942139/Pichu.Changelog.txt)

Requires these motion_list and animation changes to function properly, copy your hdr-assets folder, rename it to hdr-assets-pr and then drag and drop the contents of the zip:
[Pichu_Romfs.zip](https://github.com/HDR-Development/HewDraw-Remix/files/8941944/Pichu_Romfs.zip)

H3ra-57 committed:
Pichu Changes

H3ra-57 committed:
Pichu movement stat changes

H3ra-57 committed:
Pichu special stat changes

H3ra-57 committed:
side b stat adjustments

H3ra-57 committed:
remove motion rate on side b

SkewedAskew commented:
Some notes and suggestions:
- Jab can loop into itself quite easily as shown below (tested with DI out). I would recommend one or more of the following: increase endlag, lower angle, raise bkb
https://user-images.githubusercontent.com/55216313/174461468-645db80f-9bc9-4299-9194-4228ca49c93a.mp4
- Utilt and Dtilt should get some more bkb to avoid the same issue as jab (with kbg compensated for the increase)
- Huge fan of the Skull Bash changes. It should probably be slowed down a little bit (frame 9 -> 14-15ish) given its reward and how far Pichu moves. Also match the grounded whiff endlag to that of the aerial version (40 frames) as it is difficult to punish effectively. Skull Bash autocanceling when performed at the peak of shorthop is neat

H3ra-57 committed:
balance changes pt 1

H3ra-57 committed:
balance changes pt 2

check the updated changelog and romfs download

SkewedAskew commented:
I think the romfs might not have updated correctly to reflect the new Skull Bash startup. Early fair also may need lower kbg as it kills very early, perhaps in exchange for one or two more active frames on the late hit. Everything else feels fine

H3ra-57 committed:
pichu changes pt.3

fair no longer does your taxes (strong hit damage 13% → 12%, bkb 110 → 98, late hit lasts 3 frames longer)

SkewedAskew commented:
Lgtm

H3ra-57 committed:
Merge branch 'HDR-Development:dev' into pichu-changes

SkewedAskew commented on fighters/pichu/src/acmd/aerials.rs:
```diff
@@ -1,184 +1,181 @@
+    frame(lua_state, 9.0);
+    if is_excute(fighter) {
+        WorkModule::on_flag(boma, *FIGHTER_STATUS_ATTACK_AIR_FLAG_ENABLE_LANDING);
+        FT_ADD_DAMAGE(fighter, 2);
```
Change to decimal

SkewedAskew commented on fighters/pichu/src/acmd/aerials.rs:
```diff
@@ -1,184 +1,181 @@
+    }
+    frame(lua_state, 14.0);
+    if is_excute(fighter) {
+        FT_ADD_DAMAGE(fighter, 2);
```
Change to decimal

SkewedAskew commented on fighters/pichu/src/acmd/ground.rs:
```diff
@@ -1,9 +1,34 @@
+    let boma = fighter.boma();
+    frame(lua_state, 1.0);
+    if is_excute(fighter) {
+        FT_MOTION_RATE(fighter, 3);
```
Change to decimal

SkewedAskew commented on fighters/pichu/src/acmd/ground.rs:
```diff
@@ -1,9 +1,34 @@
+    if is_excute(fighter) {
+        ATTACK(fighter, 0, 0, Hash40::new("top"), 4.0, 70, 72, 0, 70, 3.0, 0.0, 3.0, 6.0, None, None, None, 1.0, 1.0, *ATTACK_SETOFF_KIND_OFF, *ATTACK_LR_CHECK_F, false, 0, 0.0, 0, false, false, false, false, true, *COLLISION_SITUATION_MASK_GA, *COLLISION_CATEGORY_MASK_ALL, *COLLISION_PART_MASK_ALL, false, Hash40::new("collision_attr_normal"), *ATTACK_SOUND_LEVEL_S, *COLLISION_SOUND_ATTR_PUNCH, * ...
+        ATTACK(fighter, 1, 0, Hash40::new("top"), 4.0, 70, 72, 0, 70, 3.0, 0.0, 3.0, 10.0, None, None, None, 1.0, 1.0, *ATTACK_SETOFF_KIND_OFF, *ATTACK_LR_CHECK_F, false, 0, 0.0, 0, false, false, false, false, true, *COLLISION_SITUATION_MASK_GA, *COLLISION_CATEGORY_MASK_ALL, *COLLISION_PART_MASK_ALL, false, Hash40::new("collision_attr_normal"), *ATTACK_SOUND_LEVEL_S, *COLLISION_SOUND_ATTR_PUNCH,  ...
+        FT_MOTION_RATE(fighter, 1);
```
Change to decimal

SkewedAskew commented on fighters/pichu/src/acmd/ground.rs:
```diff
@@ -1,9 +1,34 @@
+    /*frame(lua_state, 18.0);
+    if is_excute(fighter){
+        WorkModule::on_flag(boma, /*Flag*/ *FIGHTER_STATUS_ATTACK_FLAG_ENABLE_RESTART);
+    }*/
```
Remove commented code if no longer needed

SkewedAskew commented on fighters/pichu/src/acmd/other.rs:
```diff
@@ -1,90 +1,100 @@
+    if is_excute(fighter) {
+        FT_MOTION_RATE(fighter, 1.4);
+    }
+	frame(lua_state, 11.0); // Effectively F15
```
Fix indentation

SkewedAskew commented on fighters/pichu/src/acmd/other.rs:
```diff
@@ -1,90 +1,100 @@
+	frame(lua_state, 11.0); // Effectively F15
+    if is_excute(fighter) {
+		FT_MOTION_RATE(fighter, 1.0);
+		WorkModule::enable_transition_term(boma, *FIGHTER_STATUS_TRANSITION_TERM_ID_DASH_TO_RUN);
```
Fix indentation

SkewedAskew commented on fighters/pichu/src/acmd/other.rs:
```diff
@@ -1,90 +1,100 @@
+    }
+    frame(lua_state, 13.0); // Effectively F15
+    if is_excute(fighter) {
+		FT_MOTION_RATE(fighter, 1.0);
```
Fix indentation

SkewedAskew commented on fighters/pichu/src/acmd/specials.rs:
```diff
@@ -1,65 +1,129 @@
+    }
+    frame(lua_state, 18.0);
+    if is_excute(fighter) {
+        FT_MOTION_RATE(fighter, 1);
```
Change to decimal

SkewedAskew commented on fighters/pichu/src/acmd/specials.rs:
```diff
@@ -1,65 +1,129 @@
+        ArticleModule::generate_article(boma, *FIGHTER_PICHU_GENERATE_ARTICLE_DENGEKIDAMA, false, -1);
+    }
+    if is_excute(fighter) {
+        FT_ADD_DAMAGE(fighter, 1);
```
Change to decimal

SkewedAskew commented on fighters/pichu/src/acmd/specials.rs:
```diff
@@ -1,65 +1,129 @@
+    }
+    frame(lua_state, 18.0);
+    if is_excute(fighter) {
+        FT_MOTION_RATE(fighter, 1);
```
Change to decimal

SkewedAskew commented on fighters/pichu/src/acmd/specials.rs:
```diff
@@ -1,65 +1,129 @@
+        ArticleModule::generate_article(boma, *FIGHTER_PICHU_GENERATE_ARTICLE_DENGEKIDAMA, false, -1);
+    }
+    if is_excute(fighter) {
+        FT_ADD_DAMAGE(fighter, 1);
```
Change to decimal

SkewedAskew commented on fighters/pichu/src/acmd/specials.rs:
```diff
@@ -1,65 +1,129 @@
+    frame(lua_state, 4.0);
+    if is_excute(fighter) {
+        ATTACK(fighter, 0, 0, Hash40::new("top"), 14.0, 361, 71, 0, 90, 11.0, 0.0, 10.0, 0.0, None, None, None, 1.0, 1.0, *ATTACK_SETOFF_KIND_THRU, *ATTACK_LR_CHECK_F, false, 2, 0.0, 0, false, false, false, false, true, *COLLISION_SITUATION_MASK_GA, *COLLISION_CATEGORY_MASK_ALL, *COLLISION_PART_MASK_ALL, false, Hash40::new("collision_attr_elec"), *ATTACK_SOUND_LEVEL_L, *COLLISION_SOUND_ATTR_ELEC, ...
+    }
```
The above hitboxes can probably be condensed into one call

SkewedAskew commented on fighters/pichu/src/acmd/specials.rs:
```diff
@@ -1,65 +1,129 @@
+    frame(lua_state, 4.0);
+    if is_excute(fighter) {
+        ATTACK(fighter, 0, 0, Hash40::new("top"), 14.0, 361, 71, 0, 90, 11.0, 0.0, 10.0, 0.0, None, None, None, 1.0, 1.0, *ATTACK_SETOFF_KIND_THRU, *ATTACK_LR_CHECK_F, false, 2, 0.0, 0, false, false, false, false, true, *COLLISION_SITUATION_MASK_GA, *COLLISION_CATEGORY_MASK_ALL, *COLLISION_PART_MASK_ALL, false, Hash40::new("collision_attr_elec"), *ATTACK_SOUND_LEVEL_L, *COLLISION_SOUND_ATTR_ELEC, ...
+    }
```
The above hitboxes can probably be condensed into a single call

H3ra-57 committed:
Merge branch 'HDR-Development:dev' into pichu-changes

h3ra-fa2019 committed:
code cleanup

h3ra-fa2019 committed:
forgor some decimals

SkewedAskew commented:
Lgtm
